test/e2e/validation_test.go

```diff
@@ -689,3 +685,22 @@ func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatu
 	}
 	return nil
 }
+
+// getOperatorConditionName returns the operator condition name using the env var present in the deployment
+func (tc *testContext) getOperatorConditionName() (string, error) {
+	deployment, err := tc.client.K8s.AppsV1().Deployments(resourceNamespace).Get(context.TODO(), resourceName,
+		meta.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "error getting operator deployment")
+	}
+	// Get the operator condition name using the deployment spec
+	for i := range deployment.Spec.Template.Spec.Containers {
+		envVars := deployment.Spec.Template.Spec.Containers[i].Env
+		for j, each := range envVars {
+			if each.Name == condition.OperatorConditionName {
+				return envVars[j].Value, nil
+			}
+		}
+	}
+	return "", nil
```

Comment:
As the `OperatorConditionName` var is a hard requirement, I suggest returning an exception here.

Reply:
Error is returned from the method that calls getOperatorCondition name [here](https://github.com/openshift/windows-machine-config-operator/blob/f94484220f4198182a545297028b9fe10357c1c2/test/e2e/validation_test.go#L669)

Reply:
I was thinking to avoid the `ocName == ""` comparison. Unless `each.Value` could return an empty string and you are also accounting for that.